### PR TITLE
sync-diff-inspector: verify snapshot value for gc safepoint

### LINKED ISSUE #834: Sync-diff-inspector's GC keeper with specific `snapshot` does not work with TiDB v6.1 or below
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

Use TiDB as a data-source of sync-diff-inspector.  
Set `snapshot` config to some fixed value.  
Inspect the Service GC Safepoint via `pd-ctl service-gc-safepoint`

2. What did you expect to see?

The service-gc-safepoint list contains `Sync_diff_xxxxxxxxx`.

3. What did you see instead?

The safepoint is not registered.

The sync-diff log suggested the safepoint is 0.

```
[2024/12/19 16:32:30.080 +08:00] [INFO] [pd.go:228] ["generate dumpling gc safePoint id"] [id=Sync_diff_1734597150080542000]
[2024/12/19 16:32:30.080 +08:00] [DEBUG] [pd.go:230] ["update PD safePoint limit with ttl"] [safePoint=0] [updateInterval=2m30s]
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

v5.4.1, v6.1.7

5. which tool are you using?

sync-diff-inspector

6. what versionof tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

v6.5.3

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #834

### What is changed and how it works?
- verify snapshot value

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: sync_diff_inspector/diff.go [hunk 1]
```diff
@@ -322,6 +323,17 @@ func (df *Diff) compareStruct(ctx context.Context, tableIndex int) (isEqual bool
 	return isEqual, isSkip, nil
 }
 
+func verifySnap(latestSnap []string) bool {
+	if len(latestSnap) != 1 {
+		return false
+	}
+	val, err := strconv.ParseInt(latestSnap[0], 10, 64)
```
Comment (kennytm) [inline sync_diff_inspector/diff.go]: 🤔 

As `latestSnap` is a string, if the two timestamps are lexicographically reversed the `strings.Compare() < 0` below will also be wrong. Although this won't be a problem until 2090-11-18 22:07:45.625

File: sync_diff_inspector/diff_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			snapshot:       "6",
+			expected:       "5",
+		},
+		{
+			latestSnapshot: []string{"7"},
+			snapshot:       "6",
+			expected:       "6",
+		},
+		{
+			latestSnapshot:     []string{"2016-10-08 16:45:26"},
```
Comment (kennytm) [inline sync_diff_inspector/diff_test.go]: the result of SHOW MASTER STATUS should be a TSO like 454832342613098496, it should not be a datetime nor unix timestamp like 1735047693.684 🤔

Comment (GMHDBJD) [reply sync_diff_inspector/diff_test.go]: address

**Review by kennytm — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**